include/kernel/thread.h

```diff
@@ -309,3 +285,29 @@ void awaken(void);
 void postpone(sched_context_t *sc);
 #endif
 
+/* Update the kernel's timestamp and store it in ksCurTime.
+ * The difference between the previous kernel timestamp and the one just read
+ * is added to ksConsumed.
+ *
+ * Should be called on every kernel entry where threads can be billed.
+ * Will be called without holding the kernel lock!
+ */
+static inline void updateTimestamp(void)
+{
+#ifdef CONFIG_KERNEL_MCS
```

Comment:
This change makes the `updateTimestamp` function visible (and called) in all configs, and would require proof update for everything. The declaration should stay under the `CONFIG_KERNEL_MCS` guard, and the calls as well.

Might make sense to add a macro for the call that is empty (fully empty, not semicolon) when `CONFIG_KERNEL_MCS` is off to avoid lots of `ifdef` at the call sites.

Reply:
Sorry about the redo, it would add a lot of noise to the proofs. Yes, a conditionally empty macro at the call site and a conditionally included declaration/definition should work for verification (the preprocess test should then be passing for the non-MCS configs).

Reply:
I added the macro, but I didn't go for fully empty, without even semicolons added, as that seems a bit excessive. Do the proofs require updates for semicolons only? Can't those be just filtered out automatically?